content/docs/03-ai-sdk-core/15-tools-and-tool-calling.mdx

```diff
@@ -624,6 +624,87 @@ export const weatherTool = tool({
 });
 ```
 
+## MCP Tools
+
+The AI SDK supports connecting to Model Context Protocol (MCP) servers to access their tools. This enables your AI applications to discover and use tools across various services through a standardized interface.
+
+### Using MCP Tools
+
+Create an MCP client using either `stdio` (for local servers) or `SSE` (for web servers):
+
+```typescript
+const client = await createMCPClient({
+  transport: {
+    type: 'stdio',
+    command: 'node',
+    args: ['src/stdio/dist/server.js'],
+  },
+});
+```
+
+The client's `tools` method acts as an adapter between MCP tools and AI SDK tools. It supports two approaches for working with tool schemas:
+
+#### Runtime Schema Discovery
+
+```typescript
+// Use schemas provided by the server
+const tools = await client.tools();
+```
+
+#### Compile-Time Schema Definition
+
+```typescript
+import { z } from 'zod';
+
+const toolSchemas = {
+  'get-data': {
+    parameters: z.object({
+      query: z.string().describe('The data query'),
+      format: z.enum(['json', 'text']).optional(),
+    }),
+  },
+} as const;
+
+const tools = await client.tools({ schemas: toolSchemas });
+```
+
+Below is a complete example of using MCP tools with the AI SDK.
```

Comment:
instead move under Example heading and explain example there. No need to link sections with placeholders